features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/SemanticsNodeMapper.kt

```diff
@@ -9,11 +9,13 @@ package com.datadog.android.sessionreplay.compose.internal.mappers.semantics
 import androidx.compose.ui.semantics.SemanticsNode
 import com.datadog.android.sessionreplay.compose.internal.data.ComposeWireframe
 import com.datadog.android.sessionreplay.compose.internal.data.UiContext
+import com.datadog.android.sessionreplay.utils.AsyncJobStatusCallback
 
 internal interface SemanticsNodeMapper {
 
     fun map(
         semanticsNode: SemanticsNode,
-        parentContext: UiContext
+        parentContext: UiContext,
+        asyncJobStatusCallback: AsyncJobStatusCallback
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏ Why do we need this here as a function parameter ? I don't see it used.

Reply:
this parameter is assigned [here](https://github.com/DataDog/dd-sdk-android/blob/develop/features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/TreeViewTraversal.kt#L62) and called [here](https://github.com/DataDog/dd-sdk-android/blob/yl/compose/semantics-poc/features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/resources/DefaultImageWireframeHelper.kt#L96).

Briefly speaking is used to notify `recordedDataQueueHandler` asynchronously that the resource id of the image wireframe is updated.